docs(website): add Use Cases section with Light Wallets, Bridges; add Case Studies section with Daedalus (#3200)
input-output-hk/mithril: docs/website/root/use-cases/wallets.mdx

Reviewer: This usecase is about "Light Wallets", and there is another usecase for "Full Wallets" which is different (Cardano node restoration).

---
id: wallets
title: Wallets

> sidebar_label: Wallets

Reviewer 2: Should these be renamed to 'Light wallets' to address JP's comment above?

Author: Yes they should, I'll make that change

---

> # Using Mithril for Light Wallets

Reviewer 2: ```suggestion
# Using Mithril for light wallets
```

## For Wallet Developers

Cardano users expect fast, reliable confirmation that their transactions have landed. Traditionally, delivering that required either running a full node or trusting a third-party service. Mithril offers a third option: cryptographic proof of transaction inclusion, backed by Cardano's stake pool operators, that any wallet can verify independently.

:::caution

**The Mithril client library is production-ready code, but we recommend caution before deploying certificate verification in a production wallet.** Protocol security guarantees are proportional to stake participation — until the Mithril signer achieves sufficient coverage across the SPO ecosystem, there is a risk that certificates could be compromised. The greater the stake participation, the stronger the guarantees.

> :::

Reviewer: This section should be removed. We probably need to make the difference between stable/unstable features or usecases, and security of the protocol which has its own page on the website.

Author: I want to communicate to the users somehow that security guarantees are dependent on the amount of stake in the protocol. Happy to change this, remove most of it, adjust the admonition type, but not sure what the best way to communicate the caution would be.

Reviewer: The best way is to refer to the [Protocol Security](https://mithril.network/doc/mithril/advanced/mithril-protocol/security) page (for which we are working on an update with issue https://github.com/input-output-hk/mithril/issues/2703).

Author: Updated in my latest commit

### What Mithril Gives Your Wallet

When a user submits a transaction, Mithril lets you confirm it was included in the Cardano chain without running a full node and without trusting a centralized API. Instead of relying on a server telling your wallet "yes, that tx went through," you get a Merkle proof tied to a certificate signed by a quorum of SPOs weighted by stake. Your wallet verifies the proof locally.

> For users, this means faster perceived finality — you can surface a meaningful "confirmed" state sooner, and with stronger guarantees than a block count alone provides.

Reviewer 2: ```suggestion
### What Mithril provides for wallets

When a user submits a transaction, Mithril verifies that it was included on the Cardano blockchain without running a full node or relying on a centralized API. Instead of depending on a server response, the wallet receives a Merkle proof linked to a certificate signed by a stake-weighted quorum of SPOs. The wallet verifies this proof locally.

For users, this provides faster perceived finality. A wallet can present a meaningful ‘confirmed’ state earlier, with stronger guarantees than relying on block confirmations alone.
```

### How It Fits Into Your Existing Stack

Most light wallets already rely on a hosted API or indexer for balance lookups and UTxO queries — services like Blockfrost or Koios. Mithril doesn't replace that. What it adds is a trustless verification layer on top: rather than taking your data provider's word that a transaction landed, you can cryptographically confirm it against a certificate chain secured by Cardano's actual stake distribution.

This matters for security. If your data provider is compromised or returns bad data, a Mithril proof will fail to verify — your wallet catches the discrepancy rather than silently showing users incorrect state. A compromised API can't fabricate a valid Mithril proof.

> The result is an architecture that keeps what already works while meaningfully reducing the trust your users have to place in third-party infrastructure.

Reviewer 2: ```suggestion
### How it fits into existing stacks

Most light wallets already rely on a hosted API or indexer for balance lookups and UTXO queries, such as Blockfrost or Koios. Mithril does not replace these services. Instead, it adds a trustless verification layer: rather than relying on a data provider to confirm that a transaction was included, the wallet can verify this cryptographically against a certificate chain secured by Cardano’s stake distribution.

This improves security. If a data provider is compromised or returns incorrect data, a Mithril proof will fail to verify, allowing the wallet to detect the discrepancy instead of presenting an incorrect state. A compromised API cannot produce a valid Mithril proof.

The result is an architecture that preserves existing components while reducing the level of trust required in third-party infrastructure.
```

### Integration

> The easiest path for most wallet teams is the official WASM package:

Reviewer 2: ```suggestion
The easiest path for most wallet teams to integrate is using the official WASM package:
```

```bash
npm install @mithril-dev/mithril-client-wasm
```

> It works in-browser and in Node.js, covering desktop wallets, web wallets, and mobile wallets built on web technologies — iOS Safari and Android Chrome are both supported. A Rust crate is also available if you're building natively on desktop.

Reviewer: - There is also support for Firefox (you could refer to the `Browser Compatibility` section of the npm package: https://www.npmjs.com/package/@mithril-dev/mithril-client-wasm)
- We have an example for developers: https://github.com/input-output-hk/mithril/tree/main/examples/client-wasm-web
- A link to the Rust crate would be welcome: https://crates.io/crates/mithril-client

Author: I incorporated these changes, they can be viewed in the latest commit

> It works in-browser and in Node.js, covering desktop wallets, web wallets, and mobile wallets built on web technologies — iOS Safari and Android Chrome are both supported. A Rust crate is also available if you're building natively on desktop.

Reviewer 2: ```suggestion
It works in the browser and in Node.js, supporting desktop, web, and mobile wallets built on web technologies – including iOS Safari and Android Chrome. A Rust crate is also available for native desktop applications.
```

> Native mobile bindings for Swift and Kotlin are not yet available, but are on the roadmap. If you're building a native iOS or Android wallet and want to get ahead of this, reach out — the underlying C libraries are already being compiled as part of the build.

Reviewer: This is not the case, this section should be removed.

Author: Specifically you mean the section starting on line 43? Holding off on making changes on this one until I better understand what native mobile capabilities we have. I want to speak to it in some capacity given it's probably a significant developer focus.

Reviewer: Yes, there are no bindings for C/Swift/Kotlin available currently.

The core flow is straightforward:

1. Submit the transaction as you normally would
2. Query the Mithril aggregator for a transaction proof once the tx hash is known
3. Verify the proof against the current Mithril certificate chain

> 4. Surface the confirmation to the user

Reviewer 2: ```suggestion
4. Surface the confirmation to the user.
```

Author: Should I add periods to all of these items, not just #4?

> Proofs are available roughly every 30 blocks, with certification running ~100 blocks behind the tip to ensure finality. Factor this into how you communicate confirmation status to users.

Reviewer 2: ```suggestion
Proofs are generated approximately every 30 blocks, with certification operating around 100 blocks behind the chain tip to ensure finality. This should be considered when presenting confirmation status to users.
```

### Questions?

> Have a question during integration? Reach out in the [#ask-mithril](https://discord.com/channels/826816523368005654/1014019542504185876) channel on the IOG Discord — the team is active there and happy to help.

Reviewer: ```suggestion
Have a question during integration? Reach out in the [#ask-mithril](https://discord.gg/5kaErDKDRq) channel on the IOG Discord — the team is active there and happy to help.
```

<a

>   href="https://discord.com/channels/826816523368005654/1014019542504185876"

Reviewer: ```suggestion
  href="https://discord.gg/5kaErDKDRq"
```

    textDecoration: "none",
    fontWeight: "bold",
    marginTop: "4px",
    marginBottom: "1rem",
    lineHeight: "40px", // fixes vertical alignment
  }}
>
  <svg
    xmlns="http://www.w3.org/2000/svg"
    viewBox="0 0 127.14 96.36"
    width="20"
    height="20"
    fill="white"
    style={{ display: "block", flexShrink: 0 }}
  >
    <path d="M107.7,8.07A105.15,105.15,0,0,0,81.47,0a72.06,72.06,0,0,0-3.36,6.83A97.68,97.68,0,0,0,49,6.83,72.37,72.37,0,0,0,45.64,0,105.89,105.89,0,0,0,19.39,8.09C2.79,32.65-1.71,56.6.54,80.21h0A105.73,105.73,0,0,0,32.71,96.36,77.7,77.7,0,0,0,39.6,85.25a68.42,68.42,0,0,1-10.85-5.18c.91-.66,1.8-1.34,2.66-2a75.57,75.57,0,0,0,64.32,0c.87.71,1.76,1.39,2.66,2a68.68,68.68,0,0,1-10.87,5.19,77,77,0,0,0,6.89,11.1A105.25,105.25,0,0,0,126.6,80.22h0C129.24,52.84,122.09,29.11,107.7,8.07ZM42.45,65.69C36.18,65.69,31,60,31,53s5-12.74,11.43-12.74S54,46,53.89,53,48.84,65.69,42.45,65.69Zm42.24,0C78.41,65.69,73.25,60,73.25,53s5-12.74,11.44-12.74S96.23,46,96.12,53,91.08,65.69,84.69,65.69Z" />
  </svg>
  <span>#ask-mithril</span>
</a>

:::note

You'll need to be a member of the [IOG Discord server](https://discord.gg/5kaErDKDRq) to access this channel.

> :::

Reviewer: This is not needed, the `https://discord.gg/5kaErDKDRq` invitation link is already linking to the `ask-mithril` channel.

## For Wallet Users

If your wallet supports Mithril, transaction confirmations are faster and more trustworthy. Rather than relying on your wallet's servers to tell you a transaction went through, Mithril provides cryptographic proof — verified against Cardano's network of stake pool operators — that your transaction is on the chain.

In practice, this means fewer anxious minutes wondering if a payment landed, and more confidence when it says it did.

> **Does my wallet support Mithril?** Mithril integration is still being adopted across the ecosystem. Check your wallet's release notes or documentation, or ask their support team. If you're a developer building a wallet, see the section above.

Reviewer 2: ```suggestion
## For wallet users

If a wallet supports Mithril, transaction confirmations are faster and more reliable. Instead of relying on the wallet’s servers to confirm that a transaction was included, Mithril provides a cryptographic proof, verified against Cardano’s network of stake pool operators.

In practice, this reduces uncertainty around transaction status and provides greater confidence when a transaction is confirmed.

**Does my wallet support Mithril?**
Mithril integration is still being adopted across the ecosystem. Check the wallet’s release notes or documentation, or contact the support team. Developers building a wallet can refer to the section above.
```
